README.md

```diff
@@ -19,123 +19,80 @@ Polyn implements this pattern in a manner that can be applied to multiple progra
 
-### CloudEvents Extensions
+## Environment Architecture
 
-Polyn Clients MUST implement the following extensions.
+### Subscriptions
 
-#### `polyntrace`
+Polyn utilizes JetStream to distribute events to subscribing components. JetStream provides
+the ability for components to define policies around event delivery. When a Polyn application is
+deployed, a script is typically run that sets up the application component's subscriptions, much
+like a database migration.
 
-This is an array of trace objects that identify the chain of events that lead the current event to
-being published. A Polyn client SHOULD NOT fail to consume an event however if `polyntrace` is
-missing from the published event.
+### Event Lifecycle
 
-##### Example
+#### Publishing
 
-```json
-[
-  {
-    "type": "<topic>",
+When an event is published, the Polyn client validates the event against the pre-defined schema
+for that event. The schema is a [CloudEvent JSON](https://github.com/cloudevents/spec/blob/v1.0.2/cloudevents/formats/json-format.md)
+compliant [JSON Schema](https://json-schema.org/) published by the Polyn application to an internal
+repository on startup.
 
-    "time": "2018-04-05T14:31:00Z",
-    "id": "<uuid>"
-  },
-  {
-    "type": "<topic>",
-    "time": "2018-04-05T17:31:00Z",
-    "id": "<uuid>"
-  }
-]
-```
+Once validation is complete, the event is published to the bus.
 
-A Polyn client MUST provide a way to publish an event as the result of a prior event, adding the
-trace of the predecessor to the `polyntrace` array.
+#### Receiving
 
-##### Example
+When an event is received by an application, and before it is passed to the component, the event
+is again validated against the schema. Once validated it is passed to the component, where it will
+then be acknowledged as being received.
 
-```ruby
-def receive_some_event(context)
-  # ... do work
-  context.publish("<topic>", result_of_work)
-end
-```
+Once the event has been processed, it is acknowledged, making way for the next event.
 
-This would add whatever event that caused `receive_some_event` to be fired to the trace when
-`context.publish` is called.
+### Schema Repository
 
-#### `polynclient`
+When a Polyn application starts, the client publishes all of the event schema for that application
+to the schema repository. Before doing so, it compares the existing schema in the repository for
+the events of the same type, against the schema to be published, and will throw an error if the
+new schema is not backwards compatible with the old.
 
-This is an object representing the information about the client that published the event. A Polyn
-client SHOULD NOT however fail to consume the event if the `polynclient` extension is missing.
+### Error Handling
 
-##### Example
+Because components are stateless, Polyn clients adopt a "let it fail" approach to error handling.
+If an error in processing an event occurs, the clients will simply broadcast a
```

Comment:
Is this different than the `Errors` section farther down that says the service/component will `raise` if the validation fails?
Should the doc define what `data` should be included in the `polyn.error` event?